### ISSUE TITLE
Optionally store in non-Kolab friendly format

### ISSUE THREAD

**Issue by dicer** [konradrenner/kolabnotes-android#55]
I know this is a kolab client, so it makes sense how the storage format is currently defined. I personally do not use kolab, but am looking for a way to sync my notes to my IMAP server, so they are backed up and can also be viewed from a normal email client.

Unfortunately the last point is not really feasible at the moment because everything is stored in the kolab xml structure.
Would you consider implementing this when the app is out of beta and the kolab features are working as expected? Like formatting it in HTML, attaching the pictures and saving all the metadata as mail headers? Optionally of course!

**Comment by konradrenner** [comment]
Of course I consider to implement this request ;-), but first of all I am concentrating on making the app ready for a public release on Google Play.

So maybe you have to be patient because at the moment I am the only developer of the app...

**Comment by konradrenner** [comment]
Do you want to view the notes on a desktop email client or on a web client? Do you just want to view them, or also want to edit them?

**Comment by dicer** [comment]
Well, I guess viewing would be a start. In the end you will want to edit them though. More or less like messages in the draft folder. Then again, a lot of email clients make it kinda hard to actually edit things EXCEPT in the drafts folder. So not sure how useful this feature would turn out to be for most users.

**Comment by konradrenner** [comment]
Direct editing in the email is not possible. The idea is, that I maybe create a small application for desktop OSes which can sync notes. But if you want to edit notes with a web client it will not be possible.

What would be your main usecase: edit notes online via web UI or edit notes on the desktop with a client app?

Also it would be very hard to parse such an email, if there is no specific format, how the note is stored.

**Comment by kekh** [comment]
I vote up for Markdown! Especially for (live)previewing of the edited note text in the local client... The only feature that miss in this f*cking cool Notes app for Android!
You may take a look at [Turbo editor](https://github.com/vmihalachi/turbo-editor) which has a pretty good implementation for editing+live-preview.

**Comment by dicer** [comment]
I guess I would vote for web UI. It should interface with the server via IMAP though.

Regarding editing the email directly: I don't see a problem really. If it is text only or html, every email client should be able to at least view them. I would store all the other properties of the note in header fields (X-....). That's where they belong anyway. The body of the email would be the body of the note only.
It should be pretty easy to write plugins for different (desktop-)email clients which enable the user to edit the notes in a certain folder. It would basically reuse the editing features of the email client when editing and has to add some special header fields when creating new notes.
So I guess a plugin for Thunderbird would be even higher valued than a web UI by me ;)

Hope my idea is kinda clear?

**Comment by konradrenner** [comment]
@kekh I think markdown support is worth its own issue ;-). Please open a new issue and specify how you want to see this feature in the app (maybe some mockups)

**Comment by konradrenner** [comment]
@dicer the storage format and usage of mail header fields is all based on this specification: https://wiki.kolab.org/Kolab_3.0_Storage_Format

So the point is, that one should create a plugin for Thunderbird, so that Thunderbird can interpret the Kolab v3 Storage format, then it will be possible to view, edit, create, etc. notes in Thunderbird. I think that would be a really cool feature for Thunderbird!

But this has nothing to do with the kolabnotes-android app ;-), so maybe you open up an task/issue for Thunderbird developers.

By the way: It is already possible to view the notes with your local email client. Just open the "kolab.xml" attachment and you will find all informations of your note in a xml structure (modification date, color, description, etc). The subject of a mail, is the unique ID of a note.
